VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/Strings/StringPlaceholdersInWrongOrderTests.cs

```diff
@@ -0,0 +1,457 @@
+            }
+        }
+    }";
+            VerifyDiagnostic(original);
+        }
+
+        [TestMethod]
+        public void StringPlaceholdersInWrongOrder_InIncorrectOrder_WithMultipleOccurrences_InvokesWarning()
+        {
+            var original = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {1}, my name is {0}. Yes you heard that right, {0}."", ""Mr. Test"", ""Mr. Tester"");
+            }
+        }
+    }";
+
+            var expected = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {0}, my name is {1}. Yes you heard that right, {1}."", ""Mr. Tester"", ""Mr. Test"");
+            }
+        }
+    }";
+
+            VerifyDiagnostic(original, StringPlaceholdersInWrongOrderAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, expected);
+        }
+
+        [TestMethod]
+        public void StringPlaceholdersInWrongOrder_InIncorrectOrder_WithSingleOccurrence_InvokesWarning()
+        {
+            var original = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {1}, my name is {0}."", ""Mr. Test"", ""Mr. Tester"");
+            }
+        }
+    }";
+
+            var expected = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {0}, my name is {1}."", ""Mr. Tester"", ""Mr. Test"");
+            }
+        }
+    }";
+
+            VerifyDiagnostic(original, StringPlaceholdersInWrongOrderAnalyzer.Rule.MessageFormat.ToString());
+            VerifyFix(original, expected);
+        }
+
+        [TestMethod]
+        public void StringPlaceholdersInWrongOrder_InIncorrectOrder_WithUnusedPlaceholder_InvokesWarning()
+        {
+            var original = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {2}, my name is {1}. Yes you heard that right, {1}."", ""Mr. Test"", ""Mr. Tester"", ""Mrs. Testing"");
+            }
+        }
+    }";
+
+            var expected = @"
+    using System;
+    using System.Text;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+                string s = string.Format(""Hello {1}, my name is {2}. Yes you heard that right, {2}."", ""Mrs. Testing"", ""Mr. Tester"", ""Mr. Test"");
```

Comment:
This isn't correct. The placeholder starts at 1 but the used arguments start at 0.